stl/inc/__msvc_int128.hpp

```diff
@@ -719,6 +625,198 @@ struct
+        _Uint128 _Result;
+        const auto _Borrow = _SubBorrow64(0, _Left._Word[0], _Right._Word[0], _Result._Word[0]);
+        _SubBorrow64(_Borrow, _Left._Word[1], _Right._Word[1], _Result._Word[1]);
+        return _Result;
+    }
+
+    constexpr _Uint128& operator-=(const _Uint128_base& _That) noexcept {
+        const auto _Borrow = _SubBorrow64(0, _Word[0], _That._Word[0], _Word[0]);
+        _SubBorrow64(_Borrow, _Word[1], _That._Word[1], _Word[1]);
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator-=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        _Left -= _Right._Word[0];
+        return _Left;
+    }
+
+    _NODISCARD_FRIEND constexpr _Uint128 operator*(const _Uint128_base& _Left, const _Uint128_base& _Right) noexcept {
+        return _Uint128{_Uint128_base::_Multiply(_Left, _Right)};
+    }
+
+    constexpr _Uint128& operator*=(const _Uint128_base& _That) noexcept {
+        *this = *this * _That;
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator*=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        _Left *= _Right._Word[0];
+        return _Left;
+    }
+
+#if !_STL_128_DIV_INTRINSICS
+    _NODISCARD_FRIEND constexpr _Uint128 operator/(const _Uint128& _Num, const uint32_t _Den) noexcept {
+        return _Uint128{_Uint128_base::_Divide(_Num, _Den)};
+    }
+#endif // !_STL_128_DIV_INTRINSICS
+    _NODISCARD_FRIEND constexpr _Uint128 operator/(const _Uint128& _Num, const uint64_t _Den) noexcept {
+        return _Uint128{_Uint128_base::_Divide(_Num, _Den)};
+    }
+    _NODISCARD_FRIEND constexpr _Uint128 operator/(const _Uint128_base& _Num, const _Uint128_base& _Den) noexcept {
+        return _Uint128{_Uint128_base::_Divide(_Num, _Den)};
+    }
+
+#if !_STL_128_DIV_INTRINSICS
+    constexpr _Uint128& operator/=(const uint32_t _That) noexcept {
+        *this = *this / _That;
+        return *this;
+    }
+#endif // !_STL_128_DIV_INTRINSICS
+    constexpr _Uint128& operator/=(const uint64_t _That) noexcept {
+        *this = *this / _That;
+        return *this;
+    }
+    constexpr _Uint128& operator/=(const _Uint128_base& _That) noexcept {
+        *this = *this / _That;
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator/=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        if (_Right._Word[1] != 0) {
+            _Left = 0;
+        } else {
+            _Left /= _Right._Word[0];
+        }
+        return _Left;
+    }
+
+#if !_STL_128_DIV_INTRINSICS
+    _NODISCARD_FRIEND constexpr _Uint128 operator%(const _Uint128_base& _Num, const uint32_t _Den) noexcept {
+        return _Uint128{_Uint128_base::_Modulo(_Num, _Den)};
+    }
+#endif // !_STL_128_DIV_INTRINSICS
+    _NODISCARD_FRIEND constexpr _Uint128 operator%(const _Uint128_base& _Num, const uint64_t _Den) noexcept {
+        return _Uint128{_Uint128_base::_Modulo(_Num, _Den)};
+    }

```

Comment:
Unlike `operator/` above, these overloads accept `_Uint128_base`. The only `op%` overload I see below is `_Int128 operator%(_Int128, _Int128)`, so I wonder whether `_Int128 % uint32_t` and `_Int128 % uint64_t` could invoke these overloads and return `_Uint128`.

Reply:
Yip - also fixed in the next push.